src/NServiceBus.Gateway/Gateway.cs

```diff
@@ -91,14 +92,28 @@ protected override void Setup(FeatureConfigurationContext context)
             var storageConfiguration = context.Settings.Get<GatewayDeduplicationConfiguration>();
 
             context.RegisterStartupTask(b => new GatewayReceiverStartupTask(
-                channelManager, 
-                channelReceiverFactory, 
+                channelManager,
+                channelReceiverFactory,
                 GetEndpointRouter(context),
-                b.Build<IDispatchMessages>(), 
+                b.Build<IDispatchMessages>(),
                 storageConfiguration.CreateStorage(b),
-                b.BuildAll<IDataBus>()?.FirstOrDefault(), 
+                b.BuildAll<IDataBus>()?.FirstOrDefault(),
                 gatewayInputAddress,
-                context.Settings.Get<TransportInfrastructure>().TransactionMode));
+                GetTransportTransactionMode(context)));
+        }
+
+        static TransportTransactionMode GetTransportTransactionMode(FeatureConfigurationContext context)
+        {
+            try
+            {
+                return context.Settings.GetRequiredTransactionModeForReceives();
+            }
+            catch (Exception)
+            {
+                // GetRequiredTransactionModeForReceives throws on read-only endpoints.
+                // Use the transport's default mode in that case:
+                return context.Settings.Get<TransportInfrastructure>().TransactionMode;
```

Comment:
Do we have a core test to document this "backdoor"? (If not, should we add one?)

Reply:
I can check this. I think we do but I'll add a test otherwise.

Reply:
PR created here: https://github.com/Particular/NServiceBus/pull/5542